From ff3d7c9c0fc17f4fb23c172f8c8874ab4a7d6602 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 7 Oct 2016 01:43:58 +0100
Subject: [PATCH] panel: Chain methods to fix GtkBox allocation spam every half
 second

In newer versions of GTK (Notably, 3.20 onwards) we're spammed every second
or so with debug messages regarding the size allocation. In reality, this
is ticked by the clock update and any other applets requiring refresh.

As such we now chain the method to the base methods and ensure that the
internal state machine is updated accordingly, saving us from ridiculous
levels of spam in the journal and a complete inability to debug the panel.

This resolves issue #532 and it's duplicates.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 panel/panel.vala | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/panel/panel.vala b/panel/panel.vala
index b2543be..d4e4a0d 100644
--- a/panel/panel.vala
+++ b/panel/panel.vala
@@ -871,22 +871,30 @@ public class Panel : Budgie.Toplevel
 
     public override void get_preferred_width(out int m, out int n)
     {
+        int x, j;
+        base.get_preferred_width(out x, out j);
         m = scr.width;
         n = scr.width;
     }
     public override void get_preferred_width_for_height(int h, out int m, out int n)
     {
+        int x, j;
+        base.get_preferred_width_for_height(h, out x, out j);
         m = scr.width;
         n = scr.width;
     }
 
     public override void get_preferred_height(out int m, out int n)
     {
+        int x, j;
+        base.get_preferred_height(out x, out j);
         m = scr.height;
         n = scr.height;
     }
     public override void get_preferred_height_for_width(int w, out int m, out int n)
     {
+        int x, j;
+        base.get_preferred_height_for_width(w, out x, out j);
         m = scr.height;
         n = scr.height;
     }
